### PR TITLE
Housekeeping: python package deps upgrade 

### PR DESCRIPTION
**Pull request by varunshankar**
**Proposed changes**:
- confluent-kafka upgrade: https://github.com/RasaHQ/rasa/pull/12032
- pymongo upgrade: https://github.com/RasaHQ/rasa/pull/12087
- tarsafe upgrade: https://github.com/RasaHQ/rasa/pull/12089
- pytest-xdist upgrade: https://github.com/RasaHQ/rasa/pull/12132
- rocketchat upgrade: https://github.com/RasaHQ/rasa/pull/12152
- Pins tensorflow version to `2.11.0`
**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Review by Shmuma — APPROVED** [review]
Look good!

**Review by ancalita — COMMENTED** [review]
Looks good 👍🏻 The code quality failures in the tracker store module will need to be worked on, I suspect that's because of the upgrade of `pymongo`.

File: poetry.lock [hunk 1]
```diff
@@ -3113,7 +3092,7 @@ python-versions = "*"
 
 [[package]]
 name = "tensorflow"
-version = "2.11.0"
+version = "2.11.1"
```
Comment (ancalita) [inline poetry.lock:3095]: @varunshankar Judging from the CI failures for Python 3.7 on both Ubuntu and Windows, it appears that Python 3.7 support was removed from this TF micro (although their [release changelog](https://github.com/tensorflow/tensorflow/releases/tag/v2.12.0) only mentions it in the recent `2.12.0`release 😕 ) I'd advise pinning Tensorflow to `2.11.0`. If you could also submit a ticket to TF that would be great so we make them aware of this.

Comment (varunshankar) [reply poetry.lock:3095]: Thanks a lot for the help @ancalita. If I pin the tensorflow version, the `poetry update` command is stuck in resolving dependencies. It has run for an hour and its still resolving dependencies. I tried running it 2-3 times and I still face this issue.

Comment (ancalita) [reply poetry.lock:3095]: Could you check the poetry version you're using? The stuck step of resolving dependencies usually happens if you've exited some poetry command by force. You can use `poetry cache clear` [command](https://python-poetry.org/docs/cli/#cache-clear). In addition to `pypi`, I also clear `_default_cache`, but you can have a look what caches you have with `poetry cache list`. Hope it helps!

Comment (ancalita) [reply poetry.lock:3095]: After the cache clear, you can try `poetry update tensorflow` again.

**Pull request merged**